mypy.ini

```diff
@@ -0,0 +1,3 @@
+[mypy]
+
+disallow_untyped_defs = True
```

Comment:
We could probably get more aggressive on these (see the config for globus-cli for example: https://github.com/globus/globus-cli/blob/0b0565bd3124451916f744e6bc965d5d91aae0b1/setup.cfg#L15-L22 ), but we can save any updates as a future refinement.

But I think it would be better to put this into `setup.cfg` to try to stay consistent between repos.